lib/ansible/plugins/strategy/__init__.py

```diff
@@ -506,6 +506,64 @@ def normalize_task_result(self, task_result):
 
         return task_result
 
+    def search_handler_blocks_by_name(self, handler_name, handler_blocks, iterator):
+        # iterate in reversed order since last handler loaded with the same name wins
+        for handler_block in reversed(handler_blocks):
+            for handler_task in handler_block.block:
+                if not handler_task.name:
+                    continue
+                try:
+                    if not handler_task.cached_name:
+                        handler_templar = Templar(None)
+                        if handler_templar.is_template(handler_task.name):
+                            handler_templar.available_variables = self._variable_manager.get_vars(
+                                play=iterator._play,
+                                task=handler_task,
+                                _hosts=self._hosts_cache,
+                                _hosts_all=self._hosts_cache_all
+                            )
+                            handler_task.name = handler_templar.template(handler_task.name)
+                        handler_task.cached_name = True
+
+                    # first we check with the full result of get_name(), which may
+                    # include the role name (if the handler is from a role). If that
+                    # is not found, we resort to the simple name field, which doesn't
+                    # have anything extra added to it.
+                    candidates = (
+                        handler_task.name,
+                        handler_task.get_name(include_role_fqcn=False),
+                        handler_task.get_name(include_role_fqcn=True),
+                    )
+
+                    if handler_name in candidates:
+                        return handler_task
+                except (UndefinedError, AnsibleUndefinedVariable) as e:
+                    # We skip this handler due to the fact that it may be using
+                    # a variable in the name that was conditionally included via
+                    # set_fact or some other method, and we don't want to error
+                    # out unnecessarily
+                    if not handler_task.listen:
+                        display.warning(
+                            "Handler '%s' is unusable because it has no listen topics and "
+                            "the name could not be templated (host-specific variables are "
+                            "not supported in handler names). The error: %s" % (handler_task.name, to_text(e))
+                        )
+
+    def get_notified_handlers(self, notification, iterator, templar):
```

Comment:
Should we rename this?  Technically it doesn't seem to care if the handler has been notified or not, it's just getting handlers matching a name.  Or maybe just move the code to match listening handlers up into the `search_handler_blocks_by_name`?

Reply:
fwiw, when I was reading the code below at https://github.com/ansible/ansible/pull/79558/files#diff-0ec54905a2f9d2942934b11aa9b2e2c2e6184d9d8174aaf21d6ae523e48106a6R658-R663 I was really confused about the naming of `get_notified_handlers`

Reply:
I ended up doing both your suggestions, renaming `get_notified_handlers` and merging it with `search_handler_blocks_by_name` which resulted in one method `search_handers_by_notification` and cleanup up a few things along the way. I think it is nicer now.